e2e/test/MessageReceiveE2ETests.cs

```diff
@@ -184,91 +180,59 @@ public static (Message message, string messageId, string payload, string p1Value
         {
-            var wait = true;
-
             Stopwatch sw = new Stopwatch();
             sw.Start();
-            while (wait)
-            {
-                Client.Message receivedMessage = null;
 
+            while (sw.ElapsedMilliseconds < FaultInjection.RecoveryTimeMilliseconds)
+            {
                 _log.WriteLine($"Receiving messages for device {deviceId}.");
+                Client.Message receivedMessage;
+
                 if (transport == Client.TransportType.Http1)
                 {
                     // timeout on HTTP is not supported
                     receivedMessage = await dc.ReceiveAsync().ConfigureAwait(false);
                 }
                 else
                 {
-                    receivedMessage = await dc.ReceiveAsync(TimeSpan.FromSeconds(10)).ConfigureAwait(false);
+                    receivedMessage = await dc.ReceiveAsync(TIMESPAN_ONE_MINUTE).ConfigureAwait(false);
                 }
 
-                if (receivedMessage != null)
+                if (receivedMessage == null)
                 {
-                    string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
-                    _log.WriteLine($"{nameof(VerifyReceivedC2DMessageAsync)}: Received message: for {deviceId}: {messageData}");
-
-                    Assert.AreEqual(payload, messageData, $"The payload did not match for device {deviceId}");
+                    Assert.Fail($"No message is received for device {deviceId} in {TIMESPAN_ONE_MINUTE}.");
+                }
 
+                string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
+                _log.WriteLine($"{nameof(VerifyReceivedC2DMessageAsync)}: Received message: for {deviceId}: {messageData}");
+                if (Equals(payload, messageData))
+                {
                     Assert.AreEqual(1, receivedMessage.Properties.Count, $"The count of received properties did not match for device {deviceId}");
                     var prop = receivedMessage.Properties.Single();
                     Assert.AreEqual("property1", prop.Key, $"The key \"property1\" did not match for device {deviceId}");
                     Assert.AreEqual(p1Value, prop.Value, $"The value of \"property1\" did not match for device {deviceId}");
-
-                    await dc.CompleteAsync(receivedMessage).ConfigureAwait(false);
                     break;
                 }
 
-                if (sw.Elapsed.TotalMilliseconds > FaultInjection.RecoveryTimeMilliseconds)
+                try
+                {
+                    // always complete message
+                    await dc.CompleteAsync(receivedMessage).ConfigureAwait(false);
+                }
+                catch (Exception)
                 {
-                    throw new TimeoutException("Test is running longer than expected.");
+                    // ignore exception from CompleteAsync
```

Comment:
Aren't we supposed to check a specific i.e. locktoken exception?

Also ignore but just log it so will help with debugging.